schloss/consumer.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+import logging
+from socket import gaierror
+from typing import Dict, Sequence
+
+import aiokafka
+from kafka.errors import KafkaConnectionError
+
+from .dispatcher import KafkaDispatcher
+from .session import SchlossSession
+from .types import MessageTypeDispatcher, MiddlewareType
+
+logger = logging.getLogger(__name__)
+
+
+class SynchronousSchlossConsumer:
+
+    def __init__(
+            self, dependencies,
+            url: str, group_id: str,
+            handlers: Dict[str, MessageTypeDispatcher],
+            middlewares: Sequence[MiddlewareType] = (),
```

Comment:
What about accepting `dispatcher`? Or even better both variants:
```suggestion
            dispatcher: Optional[KafkaDispatcher] = None,
            *,
            handlers: Optional[Dict[str, MessageTypeDispatcher]] = 0,
            middlewares: Sequence[MiddlewareType] = (),
            …
        ):
            if dispatcher is None:
                dispatcher = …
            elif handlers or middlewares:
                raise TypeError(
                    'Either dispatcher or handlers + optional middlewares must be provided, but not both'
                ) 
```